specfile/specfile.py

```diff
@@ -826,3 +863,65 @@ def update_tag(
         )
         with self.tags() as tags:
             getattr(tags, name).value = updated_value
+
+    def update_version(
+        self,
+        version: str,
+        prerelease_suffix_pattern: Optional[str] = None,
+        prerelease_suffix_macro: Optional[str] = None,
+        comment_out_style: CommentOutStyle = CommentOutStyle.DNL,
+    ) -> None:
+        """
+        Updates spec file version.
+
+        If `prerelease_suffix_pattern` is not set, this method is equivalent
+        to calling `update_tag("Version", version)`.
+        If `prerelease_suffix_pattern` is set and the specified version matches it,
+        the detected pre-release suffix is prepended with '~' (any existing delimiter
+        is removed) before updating Version to ensure proper sorting by RPM.
+        If `prerelease_suffix_macro` is also set and such macro definition exists,
+        it is commented out or uncommented accordingly before updating Version.
+
+        Args:
+            version: Version string.
+            prerelease_suffix_pattern: Regular expression specifying recognized
+              pre-release suffixes. The first capturing group must capture the delimiter
+              between base version and pre-release suffix and can be empty in case
+              there is no delimiter.
+            prerelease_suffix_macro: Macro definition that controls whether spec file
+              version is a pre-release and contains the pre-release suffix.
+              To be commented out or uncommented accordingly.
+            comment_out_style: Whether to use `%dnl` macro or swap the leading '%'
+              with '#' to comment out `prerelease_suffix_macro`. Defaults to `%dnl`.
+
+        Raises:
+            SpecfileException if `prerelease_suffix_pattern` is invalid.
+        """
+
+        def handle_prerelease(version):
+            def update_macro(prerelease_detected):
```

Comment:
Would it be hard to factor these out? The closures are quite handy but having multiple levels makes it a bit too hard to read (at least) for me..;)

Reply:
I could make `update_macro()` be on the same level as `handle_prerelease()`, would that help?